src/main/java/frc/robot/Robot.java

```diff
@@ -25,6 +27,13 @@
 
   private RobotContainer m_robotContainer;
 
+  //network table for vision control
+  private VisionCtrlNetTable m_VisionCtrlNetTable;
+  //network table for control systems control
+  private ControlCtrlNetTable m_ControlCtrlNetTable;
+  //flag to indicate from the teleop mode
+  private Boolean m_bFromTeleMode;
```

Comment:
This variable isn't needed since you can simply call `DriverStation.getInstance().isOperatorControl()` to know if you are in TeleOp.

Reply:
I chose not to change this to the suggestion because, at the end of the match, the connection between the robot and drive station is possibly terminated, resulting in disconnection and the function not working. Also, when the robot is disabled, it is in disabled mode, so it won't call that code anymore.
We need the teleop mode before the robot is disabled because there might be a short disabled time in between the autonomous mode and the teleop mode.

Reply:
I am confused about this. Does it matter if this happens at the end of the match? The robot is disabled at the end of the match and the match is over. The DriverStation is used to operate the Robot by the DriveTeam so it should be trusted IMHO. 

Can you tell me where you have found information that leaves you with this opinion as it was the first time I have heard of this situation, not that I don't believe you, just hadn't heard from anyone over the years that this was a concern.